src/api/cpp/cvc5.cpp

```diff
@@ -5527,9 +5527,12 @@ void Solver::ensureWellFormedTerm(const Term& t) const
     bool wasShadow = false;
     if (internal::expr::hasFreeOrShadowedVar(*t.d_node, wasShadow))
     {
+      std::unordered_set<internal::Node> fvs;
+      internal::expr::getFreeVariables(*t.d_node, fvs);
       std::stringstream se;
-      se << "Cannot process term with " << (wasShadow ? "shadowed" : "free")
-         << " variable";
+      se << "Cannot process term " << *t.d_node << " with "
```

Comment:
We should not print/compute free variables if `wasShadow` is true, you should split these two cases.

Reply:
For the shadowed variables, we still need to print them. So perhaps we should add this helper function `checkVariablesInternal` to the header file
https://github.com/cvc5/cvc5/blob/main/src/expr/node_algorithm.cpp#L304
Any reason why it is not there?

Reply:
That method is internal since it includes flags that the user shouldnt have to bother setting.  The method is a generalization of other methods in the header.

If you want to report the shadowed variable in this error message, then I'd add a new method
`Node hasShadowedVar(TNode n)` which returns the variable in `n` that is shadowed.

Reply:
I would prefer a function that `getShadowedVariables`. It is not priority.